### ISSUE TITLE
Incorrect extracted text from annotation

### ISSUE THREAD

**Issue by reyjexter** [ajrcarey/pdfium-render#77]
We have an example PDF document which when using `for_annotation` to get the text, it returns weird characters. However when showing logs of page objects text, this doesn't happen.

Here's the PDF affected:

https://github.com/reyjexter/pdfium-render-wasm/blob/master/www/dynamic-template.pdf

And the example:

https://github.com/reyjexter/pdfium-render-wasm

Here's how the console log on Chrome looks like:

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/669590/225779678-f3386ec0-0da0-4ed7-b52a-df6b92340a43.png">

**Comment by ajrcarey** [comment]
Hi @reyjexter , thank you for reporting the issue.

I can reproduce the problem, but only when compiling to WASM; when running the following reproduction snippet from the console, the annotation text is returned correctly. This suggests the problem is in the WASM implementation of `FPDFText_GetBoundedText`. Checking now.

```rust
    let pdfium = Pdfium::new(Pdfium::bind_to_library(
        Pdfium::pdfium_platform_library_name_at_path("../pdfium/"),
    )?);

    let document = pdfium.load_pdf_from_file("./dynamic-template.pdf", None)?;

    for page in document.pages().iter() {
        let text = page.text()?;

        for annotation in page.annotations().iter() {
            println!(
                "Text: {:?}, bounds: {:?}",
                text.for_annotation(&annotation),
                annotation.bounds()?
            );
        }
    }
```

**Comment by ajrcarey** [comment]
Yes, the WASM implementation of `FPDFText_GetBoundedText` was the problem; the return value of `FPDFText_GetBoundedText` is the number of _characters_ (in UTF16-LE format) copied into a byte buffer, not the number of _bytes_. (This is rare difference from the usual Pdfium pattern, which is to return the number of _bytes_.) This difference was not accounted for in the WASM implementation of this function. I have pushed a fix for the problem and it will be included in crate release 0.8.0.

There will be a delay before release 0.8.0, so in the meantime you may want to set pdfium-render as a git dependency in your project.

**Comment by reyjexter** [comment]
I can confirm that the change on master works correctly on document I sent. Thanks again and we shouldn't have any issue using `pdfium-render` as a git repository.
